src/main/java/com/yahoo/bullet/querying/FilterOperations.java

```diff
@@ -0,0 +1,135 @@
+/*
+ *  Copyright 2017, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.querying;
+
+import com.yahoo.bullet.parsing.FilterClause;
+import com.yahoo.bullet.parsing.LogicalClause;
+import com.yahoo.bullet.typesystem.TypedObject;
+import com.yahoo.bullet.parsing.Clause;
+import com.yahoo.bullet.record.BulletRecord;
+
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiPredicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Stream;
+
+import static com.yahoo.bullet.common.Utilities.extractTypedObject;
+import static com.yahoo.bullet.common.Utilities.isEmpty;
+import static com.yahoo.bullet.typesystem.TypedObject.IS_NOT_NULL;
+import static com.yahoo.bullet.typesystem.TypedObject.IS_NOT_UNKNOWN;
+
+public class FilterOperations {
+    @FunctionalInterface
+    public interface Comparator<T> {
+        /**
+         * Performs the comparison operation on the {@link TypedObject} against a {@link Stream} of values.
+         *
+         * @param object The {@link TypedObject} that is the subject of the operation.
+         * @param values The {@link Stream} of values that this operation is going to performed with.
+         * @return Boolean denoting whether this comparison operation was a success or not.
+         */
+        boolean compare(TypedObject object, Stream<T> values);
+    }
+
+    // Avoids typing BiPredicate<...>
+    @FunctionalInterface
+    public interface LogicalOperator extends BiPredicate<BulletRecord, Stream<Boolean>> {
+    }
+
+    // These comparators WILL satisfy the "vacuous" truth checks. That is if the stream is empty, allMatch and
+    // noneMatch will return true; anyMatch will return false. This means that if after failing to cast all values
+    // to t's type causing the stream to be empty, the any/all/none matches will behave as above.
+    // For example:
+    // SOME_LONG_VALUE EQ [1.23, 35.2] will be false
+    // SOME_LONG_VALUE NE [1.23. 425.3] will be false
```

Comment:
It would if it were longs but these are actually all strings since our filter values are a list of strings. See the cast function in this class where the arg is a ```List<String>```. In the future, if we want to add bidirectional casting (force the data to the type of the filter value provided by the user) then we'd have to worry about this stuff.